__tests__/feature/search.test.js

```diff
@@ -0,0 +1,232 @@
+      label: 'foo',
+      type: ['http://id.loc.gov/ontologies/bibframe/AbbreviatedTitle'],
+      group: 'cornell',
+      created: '2019-11-27T19:05:52.496Z',
+      modified: '2019-11-27T19:05:52.496Z',
+    },
+  }
+
+  const bazHit = {
+    _index: 'sinopia_resources',
+    _type: 'sinopia',
+    _id: 'resource/stanford/a96f16c1-a15c-4f4f-8a25-7ed49ba1eebe',
+    _score: 0.2876822,
+    _source: {
+      title: ['baz'],
+      uri: 'http://platform:8080/resource/stanford/d7b0eb50-17bb-4258-83be-2cef2e9fc3ad',
+      label: 'baz',
+      type: ['http://id.loc.gov/ontologies/bibframe/Title'],
+      group: 'stanford',
+      created: '2019-11-27T19:05:48.496Z',
+      modified: '2019-11-27T19:05:48.496Z',
+    },
+  }
+
+  const successResult = {
+    took: 8,
+    timed_out: false,
+    _shards: {
+      total: 5,
+      successful: 5,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: { value: 3 },
+      max_score: 0.2876821,
+      hits: [fooBarHit, fooHit, bazHit],
+    },
+  }
+
+  const successResultResorted = {
+    took: 8,
+    timed_out: false,
+    _shards: {
+      total: 5,
+      successful: 5,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: { value: 3 },
+      max_score: 0.2876821,
+      hits: [bazHit, fooBarHit, fooHit],
+    },
+  }
+
+  const successResultPage1 = {
+    took: 8,
+    timed_out: false,
+    _shards: {
+      total: 5,
+      successful: 5,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: { value: 3 },
+      max_score: 0.2876821,
+      hits: [fooBarHit, fooHit],
+    },
+  }
+
+  const successResultPage2 = {
+    took: 8,
+    timed_out: false,
+    _shards: {
+      total: 5,
+      successful: 5,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: { value: 3 },
+      max_score: 0.2876821,
+      hits: [bazHit],
+    },
+  }
+
+  const noResult = {
+    took: 8,
+    timed_out: false,
+    _shards: {
+      total: 5,
+      successful: 5,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: { value: 0 },
+      max_score: 0,
+      hits: [],
+    },
+  }
+
+  // Saves global fetch in order to be restored after each test with mocked fetch
+  const originalFetch = global.fetch
+
+  afterEach(() => {
+    global.fetch = originalFetch
+  })
+
+
+  it('allows the user to filter results', async () => {
+    global.fetch = jest.fn().mockImplementation(() => Promise.resolve({ json: () => successResult }))
+
+    renderApp()
+
+    fireEvent.click(screen.getByText(/Linked Data Editor/, { selector: 'a' }))
+    fireEvent.click(screen.getByText(/Search/, { selector: 'a' }))
+    fireEvent.change(screen.getByLabelText('Query', { selector: 'input#searchInput' }), { target: { value: '*' } })
+    fireEvent.click(screen.getByLabelText('Submit search', { selector: 'button' }))
+
+    await screen.findByText(/foo bar/)
+
+    // TODO: why don't filtering options show up in test UI?
```

Comment:
using `screen.debug()`, afaict, the filter elements were not rendering in the page for the tests.  i haven't figured out why, though i suspect maybe my fixture search results are incomplete?  but i didn't see anything obviously related missing from them, compared to what i saw when i ran searches on my local instance and looked at the response traffic in the browser's network debugger.

Reply:
Created issue https://github.com/LD4P/sinopia_editor/issues/2499